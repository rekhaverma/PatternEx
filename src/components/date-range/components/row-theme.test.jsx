import React from 'react';
import moment from 'moment';
import RowTheme from './row-theme.component';

/**
 * @status: Complete
 * @sign-off-by: Alex Andries
 */
describe('<RowTheme />', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      startDate: moment.utc('2018-01-01'),
      endDate: moment.utc('2018-01-07'),
      className: 'row-theme',
      onClick: jest.fn(),
    };
    wrapper = shallow(<RowTheme {...props} />);
  });

  it('should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onClick when element is clicked', () => {
    wrapper.find('.row-theme').first().simulate('click');
    expect(props.onClick.mock.calls.length).toBe(1);
  });
});
