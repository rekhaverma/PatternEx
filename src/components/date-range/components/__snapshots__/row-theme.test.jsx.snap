// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RowTheme /> should match with snapshot 1`] = `
<div
  className="row-theme"
  onClick={[Function]}
>
  <div
    className="row-theme__row"
  >
    <FormattedHTMLMessage
      id="dateRange.row.fromToUntil"
      values={
        Object {
          "valueFrom": "01 - 01 - 2018",
          "valueUntil": "01 - 07 - 2018",
        }
      }
    />
    <span
      className="icon-Calendar-icon"
    />
  </div>
</div>
`;
